platform/spire/src/setup.py

```diff
@@ -108,6 +108,10 @@ def ssh_mkdir(self, name: str, node: configuration.Node, *paths: str, with_paren
         options = ["-p"] if with_parents else []
         self.ssh(name, node, "mkdir", *(options + ["--"] + list(paths)))
 
+    def ssh_rm(self, name: str, node: configuration.Node, path:str, allow_recursive: bool=False) -> None:
+        options = ["-rf"] if allow_recursive else ["-f"]
+        self.ssh(name, node, "rm",  *(options + ["--"] + [path]))
```

Comment:
Is there a reason we need `allow_recursive` as an argument? I don't see it used anywhere.

Reply:
We do not need the `allow_recursive` argument right now. I added it since I figured it may be useful later, but for now, I have removed it in the rewrite of this change after the rebase.